src/main/java/hu/mta/sztaki/lpds/cloud/simulator/helpers/trace/TraceProducerFoundation.java

```diff
@@ -77,4 +80,44 @@ public TraceProducerFoundation(final Class<? extends Job> jobType) throws NoSuch
 	public long getMaxProcCount() {
 		return maxProcCount;
 	}
+
+
+	/**
+	 * Performs the {@link #getAllJobs()} function and then sorts the collection using a Job Comparitor
+	 *
+	 * @param jobComparator - The comparitor used to perform the sorting
+	 * @return - If there were no previous reading of the tracefile by this
+	 *         reader, then a sorted set of jobs in the range between "from" and
+	 *         "to". Otherwise a null list is returned.
+	 */
+	@Override
+	public List<Job> getAllJobs(Comparator<Job> jobComparator) throws TraceManagementException {
+		List<Job> jobList = getAllJobs();
+
+		if(jobList == null)
+			return null;
+
+		Collections.sort(jobList, jobComparator);
+		return jobList;
+	}
+
+	/**
+	 * Performs the {@link #getJobs(int)} function and then sorts the collection using a Job Comparitor
+	 *
+	 * @param num the number of jobs to be collected in the current run.
+	 * @param jobComparator - The comparitor used to perform the sorting
+	 * @return the sorted set of jobs collected from the tracefile.
+	 * @throws NoFurtherJobsException
+	 *             if there are no further jobs available in the tracefile.
+	 */
+	@Override
+	public List<Job> getJobs(int num, Comparator<Job> jobComparator) throws TraceManagementException {
+		List<Job> jobList = getAllJobs();
+
+		if(jobList == null)
```

Comment:
This is duplicate code, move it to a function of its own and refactor so getAllJobs and getjobs use the same sorting mechanism.